### ISSUE TITLE
Licensing implications needs to be determined

### ISSUE THREAD

**Issue by saschaelble** [saschaelble/UE4_Environment_Community_Project#1]
Because this is combined code from two projects, there are some considerations. (Also author permission to change or modify the license to prevent the issue described below)

I have a problem with someone thinking they can grab a bunch of "dead" community projects, update them, and then merge them into one plugin, so they can SELL it. FORCING users of this FREE COMMUNITY ASSET to BUY your "updated" version. Just so they can basically use the same FREE version, but on a newer engine version. (or have a bug fix no one has been able to figure out yet)

In other words : PLUGIN UPDATES FOR MONEY ARE FORBIDDEN
Okay for use in commercial games/software ect.

Don't think just because you throw in a few extra features on your plugin that I'm fooled.

A particular case, was the work done by Johnathan. While no one had a problem with him charging for his plugin (he created a lot of content/features of his own work), I think the community reported him because he fixed stuff in the original code, and never shared it. Basically violating the spirit of the project.

We don't mind you taking pieces and creating your own, and combining them, and creating something for sale on the market place, but we do mind when you take our pieces and bug fix them, and keep them hidden from the community for the benefit of your pocket.

We are not asking that you share your part of the code, your added features or major code changes.

**Comment by NilsonFLima** [comment]
The problem with Johnathan ocean, which I have purchased and got auto refund from Epic when the asset got removed, was that he copied content from the Physical Water Surface from Theokoles: basically the ocean plane, bottom and the buoyancy component. The blueprint components were totally copied 100%. I have asked him myself when the asset got removed and he did tell me there were nothing else, just to discover few posts from him later (which he deleted from forum btw, but the traces from some can be found because some people quoted him) where he were asking things, got directions and implemented following work from people in the WIP Community Ocean. From what I have been told, he claimed to Epic that those parts were made by him, and this granted not only the asset removal, but also his seller account to be blocked (that username can't sell anymore).

**Comment by NilsonFLima** [comment]
About the license used by the original author, he has choosen MIT, so every derived work must contain his authorship in the code, thats all. Every piece of code not in the original, must have a separate copyright notice with attribution of the author and then this author can follow the same MIT license or use something different only on his own portions. MIT as it is, allows to you sell the code, but if it contains something in a different license it cannot. I will check the other options and post here if there is any suitable for the job.

**Comment by saschaelble** [comment]
It's not really that big of a deal as long as unreal is willing to step in when needed...
I have no problem with as is

**Comment by NilsonFLima** [comment]
I have a response from Epic's Legal department about two projects: VaPlugin and Realistic Ocean Simulator.
Both projects contains code which belongs to NVidia and both have excluded the copyright notice from NVidia and replaced with their own.
VaPlugin: author claim it is his copyright, but offer it as MIT license, but uses a lot of material from NVidia
Realistic Ocean Simulator: they claim it is their copyright only, but uses some of material from NVidia.

They have provided the following text: 
===================================================================
"It was determined that the MIT code was not substantial enough to be a legal concern, however, there is an issue with the Nvidia EULA which is in direct violation of the Marketplace Distribution Agreement and Marketplace Guidelines.

We have made the product temporarily unavailable for purchase so that the seller may make the necessary changes to bring their product into legal compliance. We have contacted the seller asking them to bring the product into compliance by removing or replacing the offending source code within the next 60 days. Once the changes have been made we will republish the product. If the changes are not completed within 60 days, further action will be taken."
===================================================================

Basically, anyone can have the code from VaPlugin and use it into their project, but as it has code which violates NVidia's Eula, it is not possible legally to sell it, since it does not belong to anyone else. This is good, means everyone can use this ocean implementation for a project, but it cannot be packaged for reselling at Marketplace as a plugin.

So we can move along to make this VaPlugin part of the FREE project!!

**Comment by saschaelble** [comment]
Aww I wasn't trying to knock someone else down. But taking free code and
selling it IS kinda shitty so whatever.

**Comment by NilsonFLima** [comment]
Well, too many people see code on the Internet and think because it is there, it might be free. It is necessary a keen eye on license and copyright notices, keep record of those notes if they are in the Internet and go to a Notes Office to register how that text was when you used it, because text over the internet can change over time.
